src/custom/hooks/useWalletInfo.ts

```diff
@@ -55,12 +71,18 @@ export function useWalletInfo(): ConnectedWalletInfo {
     }
   }, [connector])
 
+  useEffect(() => {
+    if (account && web3Instance) {
+      checkIsSmartContractWallet(account, web3Instance).then(setIsSmartContractWallet)
```

Comment:
nitpicking, we could store it in the localstore, since this won't change ever and the user will likelly be comming back to cowswap or refreshing the UI

Reply:
Hmmm true.
Although, that means redux boilerplate, which I don't wanna get anywhere near =P

But maybe in another PR.

Reply:
no no, I would keep redux out of this :) 
It could be within the hook, right?

Reply:
:relieved: 

Yes, better idea, definitely.
Will give this a try.

Reply:
One problem that I think we can have is with Pillar.
When the app is created, it's just an EOA address.
Once you try do your first transaction, you are prompted to "activate" the wallet.
That requires funds and it'll actually deploy the SC wallet at that same address.

So, if you have connected with Pillar before "activating" it, the wallet will forever be considered NOT a SC wallet.

And Pillar provides no info with WC metadata, so we can't tell it's a Pillar wallet :disappointed: 

I'll leave it as is for this reason

Reply:
okok